editor/store/effects.js

```diff
@@ -303,6 +307,9 @@ export default {
 				return block;
 			} );
 			effects.push( resetBlocks( blocks ) );
+		} else if ( getDefaultBlockForPostFormat( post.format ) ) {
+			setDefaultBlockName( getDefaultBlockForPostFormat( post.format ) );
```

Comment:
It's a bit odd to me that we assign the default block into the registry from the editor initialization, and makes me wonder if default block is not best represented by the registry / API and instead by the editor itself.